engine/src/repository/game.rs

```diff
@@ -18,20 +20,30 @@ pub enum GameError {
     CreateGameFailed,
 }
 
-impl LaerningToolRepository {
-    pub async fn create_game(&self, game: Game) -> Result<Game, GameError> {
+#[async_trait]
+impl Repository<Game, GameError> for LaerningToolRepository {
+    async fn create(&self, game: Game) -> Result<Game, RepositoryError<GameError>> {
```

Comment:
We would like to have entities in repository names. Otherwise there will be clashes with `create_dataset`, `create_game`, `create_user` etc.
```suggestion
    async fn create_game(&self, game: Game) -> Result<Game, RepositoryError<GameError>> {
```

Reply:
It actually won't clash because the Trait takes a Generic type, so whatever time is implied would be the type there.

But would revert back the original names and find a way round it

Reply:
a typical example of this type of trait implementation is Add trait [https://doc.rust-lang.org/std/ops/trait.Add.html], so irrespective of the type it won't clash.